### PR TITLE
fix Test_GetLogsForQuery

### LINKED ISSUE #1835: fix Test_GetLogsForQuery
# fix Test_GetLogsForQuery, TestFilterWebsocket and TestFilterPendingTxWebsocket
## Description

Currently, we are checking the length of the logs when there is an error. Instead, we should check for the length of the logs where there is no error.

[`https://github.com/0xPolygon/polygon-edge/blob/25e09c2fcc4083a302165d461082b4456a5f3634/jsonrpc/filter_manager_test.go#L150C1-L150C1`](https://github.com/0xPolygon/polygon-edge/blob/25e09c2fcc4083a302165d461082b4456a5f3634/jsonrpc/filter_manager_test.go#L150C1-L150C1)

For TestFilterWebsocket and TestFilterPendingTxWebsocket
- Accommodate review changes suggested in #1787

### PR DESCRIPTION
**Pull request by sarvalabs-shravanth**
# Description

Resolves #1835 

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [x] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

Please complete this section if you ran manual tests for this functionality, otherwise delete it

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

**Comment by sarvalabs-shravanth** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by Stefan-Ethernal — APPROVED** [review]

**Comment by Stefan-Ethernal** [comment]
Hey @sarvalabs-shravanth, tnx for the PR. Notice that linting is failing, so just please fix it and we are good to go with merging it after you gain one more approval 🙂

**Review by goran-ethernal — APPROVED** [review]

**Review by goran-ethernal — APPROVED** [review]

**Review by vcastellm — APPROVED** [review]
LGTM thx for the contribution!

**Pull request merged**
